.github/workflows/daily-tests.yaml

```diff
@@ -172,6 +172,57 @@ jobs:
         rm -rf ~/.cache || true
     - run: echo "This job's status is ${{ job.status }}."
 
+  # split library example tests into runs based on Suite UID
+  # so that they don't hog the runners for too long
+  testlib-long-gem5_library_example_tests_x86_ubuntu_run_ALL_x86_64_opt:
+    runs-on: [self-hosted, linux, x64, run]
+    strategy:
+      matrix:
+        test-type: [x86_ubuntu_run_ALL_x86_64_opt, riscv_ubuntu_run_ALL_x86_64_opt, lupv_example_ALL_x86_64_opt, arm_ubuntu_run_test_ALL_x86_64_opt, riscvmatched_hello_ALL_x86_64_opt]
+    container: gcr.io/gem5-test/ubuntu-22.04_all-dependencies:latest
+    needs: build-gem5
+    timeout-minutes: 1440 # 24 hours
+    steps:
+    - uses: actions/checkout@v3
+      with:
+        # Scheduled workflows run on the default branch by default. We
+        # therefore need to explicitly checkout the develop branch.
+        ref: develop
+    - uses: actions/download-artifact@v3
+      with:
+        name: ${{needs.name-artifacts.outputs.build-name}}ALL
+        path: build/ALL
+    - run: chmod u+x build/ALL/gem5.opt
+    - uses: actions/download-artifact@v3
+      with:
+        name: ${{needs.name-artifacts.outputs.build-name}}ALL_MESI_Two_Level
+        path: build/ALL_MESI_Two_Level
+    - run: chmod u+x build/ALL_MESI_Two_Level/gem5.opt
+    - name: long ${{ matrix.test-type }} gem5_library_example_tests
+      working-directory: ${{ github.workspace }}/tests
+      run: ./main.py run --uid SuiteUID:tests/gem5/gem5_library_example_tests/test_gem5_library_examples.py:test-gem5-library-example-${{ matrix.test-type }} --length=long --skip-build -vv
```

Comment:
I don't like how brittle this is, as if we rename any of these directories or even the test, things will break, but I can't see a better way to do it, so I won't block on this. It's just something to keep in mind.